pulp_rpm/app/modulemd.py

```diff
@@ -65,15 +65,17 @@ def split_modulmd_file(file):
     m_open = (
         gzip.open if file.url.endswith(".gz") else lzma.open if file.url.endswith(".xz") else open
     )
-    with m_open(file.path, "r") as modulemd_file:
+    with m_open(file.path, "rb") as modulemd_file:
```

Comment:
have you had chance to test with another compress method for example `.gz` from fedora modular?

Reply:
just noted test `test_modular_yaml` using repository with gzipped modules

Reply:
Yeah - I ran all our tests, and the performance tests, and they all worked. That exercises a variety of repositories.

When debugging this particular failure, what confused me completely is that the decode() works for every line of the file **except for the last one** . Wth?

Reply:
When you say the decode doesn't work, what do you mean?

Reply:
`'str' object has no attribute 'decode'`

`line` is a byte-object, that needs to be decode()ed to str - until suddenly it isn't.

Reply:
> When debugging this particular failure, what confused me completely is that the decode() works for every line of the file **except for the last one** . Wth?

Also, to get this recorded for posterity before I forget - this is not quite true. The point at which we fail is a module-separator **in the middle of modulemd**. The parse has successfully `decode()`ed ~39K lines before failing, roughly mid-file.